src/Agent.Listener/CommandSettings.cs

```diff
@@ -163,6 +164,31 @@ public bool GetRunAsService()
                 defaultValue: false);
         }
 
+        public bool GetEnableAutoLogon()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.EnableAutoLogon,
+                description: StringUtil.Loc("EnableAutoLogon"),
+                defaultValue: false);
+        }
+
+        public bool GetRestartNow()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.RestartNow,
+                description: StringUtil.Loc("RestartNow"),
+                defaultValue: true);
+        }
+
+        public string GetAutoLogonUserName()
```

Comment:
Can you remove this method and reuse GetWindowsLogonPassword and pass a bool if you need a different string